### ISSUE TITLE
How can I perform  subscribe of a new user

### ISSUE THREAD

**Issue by maksmachado** [nbd-wtf/nostr-tools#230]
Hello friends, I'm starting to study nostr-tool. I would like to know how do I register a new user on the network!

**Comment by maksmachado** [comment]
Yes, I was referring specifically to sending an event with user metadata. However, I only found one example:

{
name:
about:
picture
}

However, this example does not make it clear how to record the "@" symbol and the display name of the user.

**Comment by maksmachado** [comment]
Sorry, my question wasn't clear, so I tested some clients and realized that they were able to track both the username and the display name, which are usually different. In the example I provided, it only shows one property, "name." So, I thought that maybe the complete way to do this would be:

{
name:
display_name:
}

Or something like that.

**Comment by maksmachado** [comment]
great. so I tested some clients and realized that they were able to track both the username and the display name, which are usually different. In the example I provided, it only shows one property, "name." So, I thought that maybe the complete way to do this would be:

{
name:
display_name:
}

or something like that

**Comment by maksmachado** [comment]
i have tried to perform this:

`const privateKey = NostrTools.generatePrivateKey();
            navigator.clipboard.writeText(privateKey);
            const publicKey = NostrTools.getPublicKey(privateKey);

            const pool = new NostrTools.SimplePool();
            const event = {
                kind: 0,
                pubkey: publicKey,
                created_at: Math.floor(Date.now() / 1000),
                name: userName
                
            }

            const standardePicture = 
           "https://gateway.pinata.cloud/ipfs/QmYHfSFLAZvKR6c11xDFbc3W25Q3SWcpcRtZ1c4RQ6emdZ"

            event.about = about === '' ? `standard about` : about.value;
            event.picture = picture === '' ? standardePicture : picture.value;
            if(displayName.value !== '') event.display_name = displayName.value;

            event.id = NostrTools.getEventHash(event);
            event.sig = getSignature(event, privateKey);

            const pub = pool.publish(relayList, event);`

but a get this error: can't serialize event with wrong or missing properties

**Comment by maksmachado** [comment]
i've tried to perform the code below:

`const privateKey = NostrTools.generatePrivateKey();
            const publicKey = NostrTools.getPublicKey(privateKey);

            const pool = new NostrTools.SimplePool();
            const event = {
                kind: 0,
                pubkey: publicKey,
                created_at: Math.floor(Date.now() / 1000),
                name: userName.value
                
            }

            const standardPicture = `https://ex.com/picture`;

            event.about = about.value === '' ? `Um feliz habitante de Pangeia` : about.value;
            event.picture = picture.value === '' ? standardPicture : picture.value;
            if(displayName.value !== '') event.display_name = displayName.value;

            console.log(event);
            

            event.id = NostrTools.getEventHash(event);
            event.sig = NostrTools.getSignature(event, privateKey);

            const pub = pool.publish(relayList, event);`

but i've get thi error:
can't serialize event with wrong or missing properties

**Comment by alexgleason** [comment]
It's missing `tags` and `created_at`

**Comment by maksmachado** [comment]
I placed an empty array in the "tag" property, but the error persists.

any idea?

**Comment by alexgleason** [comment]
It's missing `content`. Look at NIP-01 again

**Comment by maksmachado** [comment]
Now I get it, I was thinking that standard was only needed for kind 1 events.

thanks!

**Comment by maksmachado** [comment]
I am trying to fetch using this code:

`const pool = new NostrTools.SimplePool();
    const publicKey = localStorage.getItem("publicKey");
    console.log(publicKey);

    const sub = pool.sub([...relayList], [
        {
            kinds:[0],
            authors: [publicKey]
        }
    ]);
    
    
    sub.on('event', event => {
        console.log(event);
    });`

but is returning another event
